ARE/middleware/src/main/java/eu/asterics/mw/are/BundleManager.java

```diff
@@ -647,84 +805,57 @@ public void installServices() {
+        StringBuilder bundleDescriptorsXMLBuilder=new StringBuilder();
+        bundleDescriptorsXMLBuilder.append("<?xml version=\"1.0\"?>");
+        bundleDescriptorsXMLBuilder.append("<componentTypes xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\">");
 
-                URL bundleDescriptor = new URL(inputFilePath);
-                URI relativeURI = ResourceRegistry.getInstance().toRelative(componentJarURI);
+        List<Bundle> installableBundleList=getInstallableBundleList();
 
-                Set<IComponentType> componentTypeSet = DefaultBundleModelParser.instance
-                        .parseModel(bundleDescriptor.openStream());
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(componentList))) {
+            for(Bundle bundle : installableBundleList) {
+                URL bundleDescriptor=bundle.getResource(DefaultBundleModelParser.BUNDLE_DESCRIPTOR_RELATIVE_URI);
+
+                String bundleDescriptorAsXMLString=ResourceRegistry.getResourceContentAsString(bundleDescriptor.openStream());
+                Set<IComponentType> componentTypeSet = DefaultBundleModelParser.instance.parseModelAsXMLString(bundleDescriptorAsXMLString);
+                
+                //update big bundle descriptors string 
+                bundleDescriptorsXMLBuilder.append(removeSurroundingComponentTypesTags(bundleDescriptorAsXMLString));
+                String installableBundleJarName=ResourceRegistry.getInstance().toRelative(bundle.getLocation()).getPath();
+                
+                //extract component types and store them in loader_componentlist.ini
                 if (componentTypeSet.size() >= 1) {
-                    StringBuffer line = new StringBuffer(relativeURI.getPath());
+                    StringBuilder line = new StringBuilder(installableBundleJarName);
                     for (IComponentType componentType : componentTypeSet) {
                         line.append(COMPONENTLIST_DELIM);
                         line.append(" ");
                         line.append(componentType.getID());
-
-                        if (!ResourceRegistry.getInstance().isOSGIMode()) {
-                            // Also install ComponentType in
-                            // ComponentRepository, to support non-osgi mode.
-                            // This is actually not a good location for doing it
-                            // because the main purpose of this method is to
-                            // generate the cache, but
-                            // here we have all we need.
-                            ComponentRepository.instance.install(componentType);
-                        }
                     }
                     writer.write(line.toString());
                     writer.newLine();
                 }
             }
             writer.flush();
             writer.close();
```

Comment:
i think you do not need these lines anymore because you used an try-with-resource statement which does closing automatically, see https://docs.oracle.com/javase/7/docs/technotes/guides/language/try-with-resources.html

Reply:
correct